image_tools/src/cam2image.cpp

```diff
@@ -113,9 +114,10 @@ int main(int argc, char * argv[])
   // parameter.
   custom_camera_qos_profile.history = history_policy;
 
+  std::cout << "Publishing data on topic " << topic << std::endl;
```

Comment:
I think it depends on the implementation of the C library. For POSIX the following is stated in http://pubs.opengroup.org/onlinepubs/9699919799/functions/flockfile.html :

> All functions that reference (FILE *) objects, except those with names ending in _unlocked, shall behave as if they use flockfile() and funlockfile() internally to obtain ownership of these (FILE *) objects.

I don't think I have ever seen a single `printf` statement being interleaved with another `printf` statement from a different thread. For `cout << "foo" << "bar" << endl` this will certainly be written in three separate steps and interleaving is quite common for those.